src/MassTransit.Abstractions/JobService/Contracts/JobService/JobSubmitted.cs

```diff
@@ -23,6 +23,11 @@ public interface JobSubmitted
         /// </summary>
         TimeSpan JobTimeout { get; }
 
+        /// <summary>
+        /// An optional concurrency limitation key. Job types with the same key will not run concurrently 
+        /// </summary>
+        string ConcurrencyKey { get; }
```

Comment:
May be worth introducing a value type to strongly type `ConcurrencyKey` instead o relying on string. Then could avoid the need for `string.IsNullOrEmpty` later

Reply:
What do you suggest? `Guid`? 
I felt that string provides the most flexibility, especially if you want to provide a key like `5715424725204_001_B2B`. How would you do that as a `Guid`? Sure there's probably libraries that can come up with unique `Guid`s for that, but it seems like a lot of extra effort for no reason (other than the _tiny_ cost saving of comparing value types instead of comparing strings)

Reply:
I'd just create my own class/ValueType called `CurrencyKey` which could accept a string, but it abstracts away the null checking and such. You could have a UUID based approach [using a deterministic UUID](https://github.com/Informatievlaanderen/deterministic-guid-generator), though would still require a seed to produce it properly.

It's only minor, and if it's just one `IsNullOrEmpty` check it's probably fine to leave it as is. If you use a class for `CurrencyKey` then it could be backed by any type of value you want behind the scenes too, if that helped optimize anything further

Reply:
Since it needs to be inserted in the database, it would have to resolve to a string at that point in time anyways, in which case I think it is not worth the effort/complication 😅 

I think the main concern for me is that someone uses huge strings for this (as this would take up more database space and increase comparison time). Solution: added to the doc that one shouldn't do that

Reply:
No one reads the docs 😁 Also it's fine to keep it as is too.

That's my point though, it gives you the opportunity to optimize what it really is stored as. So maybe you could take any string in, but use xxHash to persist it as a smaller string. Using a deterministic UUID would allow you to define the type DB side as `uuid` too.

You'd lose the ability to easily see what the original key was, if that is required.